src/controllers/session.ts

```diff
@@ -0,0 +1,60 @@
+import { Response, NextFunction } from 'express';
+import { RequestWithUserRole, AppointmentWithTherapistOptional } from '../types';
+import generateMeeting from '../services/calendar';
+import { templateErrors } from '../helpers';
+import { BookedAppointment } from '../services';
+import { User } from '../models';
+
+const createSessionController = async (
+  req: RequestWithUserRole,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const userData = req.user;
+    const { appointmentId } = req.body;
+
+    const appointment: AppointmentWithTherapistOptional
+    | null = await BookedAppointment(appointmentId);
+
+    if (!appointment) {
+      throw templateErrors.BAD_REQUEST('The requested appointment is not available.');
+    }
+    if (!userData) {
+      throw templateErrors.UNAUTHORIZED('user not found');
+    }
+
+    appointment.isBooked = true;
+    await appointment.save();
+
+    const therapistEmail = appointment?.therapist?.user?.email;
+    if (!therapistEmail) {
+      throw templateErrors.BAD_REQUEST('The Therapist email is not available');
```

Comment:
how it's possible not to find the therapist's email if it likely it will be a server error not a bad request